### PR TITLE
Add support for mapping data retrieved with topXdata query to pandas DataFrames

### PR DESCRIPTION
**Pull request by mmac-m3a**
This PR adds support for mapping data retrieved with `topXdata` query to pandas DataFrames.
This functionality is provided by the `DataQueryDefinition` class.

### CODE REVIEW

File: kentik_api_library/kentik_api/analytics/README.md [hunk 1]
```diff
@@ -7,22 +7,126 @@ is installed with the `analytics` option (`kentik-api[analytics]`).
...
+want to insert beginning of the queried interval, include `{start_time}` placeholder where the value is expected in
+the query and then provide `start_time=<desired_start_time>` among arguments to the `{Data|SQL}MappedQuery.get_data()`
+method. Any value that can be expanded using `str.format` syntax can be interpolated at runtime.
+
+#### Response mapping
+Result mapping represented by `ResultMapping` class is (basically) a dictionary keyed by output DataFrame columns names
+with entries providing data for constructing values of the column from Kentik API query responses. Individual
+mapping entries are represented by the `MappingEntry` class. Mapping entries contain following attributes:
+- `source`: string containing `str.format`-style formatting string for extracting specific result data fields. Special
+          directives, prefixed with @TS allow extracting data from `topXdata` result `timeSeries` blocks (and hence are
```
Comment (mateuszmidor) [inline kentik_api_library/kentik_api/analytics/README.md:29]: "in the context of" probably?

File: kentik_api_library/kentik_api/analytics/flatness.py [hunk 2]
```diff
@@ -357,22 +356,29 @@ def flatness_analysis(
...
                     column of the "data" DataFrame
-    :param data: DataFrame indexed by time ("ts" column) containing columns "link" and "bytes_out". The "link" column is
-                 expected to contain names of network links as <device_name>:<interface_name> and :"bps_out" contains
-                 sum of outbound traffic rate (in bits per second) for the interval ending at the row timestamp
+    :param data: DataFrame indexed by time ("ts" column) containing columns "link" and "bytes_out" or "bps_out.
+                 "link" column is expected to contain names of network links as <device_name>:<interface_name>
+                 "bytes_out" columns (if present) is expected to contain total number of bytes transmitted via the link
+                 "bps_out" column (if present) is expected to contain average outbound bandwith in bits/s for the link
+                 For "bytes_out" and "bps_out" the value is for a time period ending at the timestamp.
+                 If the "bps_out" column is present it used as the source for the analysis (regardless of presence of
```
Comment (mateuszmidor) [inline kentik_api_library/kentik_api/analytics/flatness.py:364]: "it is used" probably

**Review by mateuszmidor — APPROVED** [review]

File: kentik_api_library/kentik_api/analytics/README.md [hunk 3]
```diff
@@ -7,22 +7,126 @@ is installed with the `analytics` option (`kentik-api[analytics]`).
...
+          applicable only the context of `DataMappedQuery`)
```
[the remark this answers is not in the document]
Comment (mmac-m3a) [reply kentik_api_library/kentik_api/analytics/README.md:30]: Yes.Fixing.

File: kentik_api_library/kentik_api/analytics/flatness.py (hunk 2 above)
Comment (mmac-m3a) [reply kentik_api_library/kentik_api/analytics/flatness.py:364]: Yes.

**Pull request merged**